ui/src/pages/namespace/Explorer/Tree/index.tsx

```diff
@@ -44,7 +45,23 @@ const ExplorerPage: FC = () => {
   const [deleteNode, setDeleteNode] = useState<BaseFileSchemaType>();
   const [renameFile, setRenameNode] = useState<BaseFileSchemaType>();
   const [previewNode, setPreviewNode] = useState<BaseFileSchemaType>();
+
+  const [search, setSearch] = useState("");
   const { t } = useTranslation();
+  const isSearch = search.length > 0;
+
+  const children = useMemo(
+    () => (data?.type === "directory" && data?.children) || [],
+    [data]
+  );
+
+  const filteredFiles = useMemo(
+    () =>
+      (children ?? [])?.filter(
```

Comment:
```suggestion
      children.filter(
```
I get no linting errors with this. Not sure I understand what you were trying to do here, but doesn't line 54 already determine that there is at least an empty array?